src/linear/updater_coordinate.cc

```diff
@@ -0,0 +1,133 @@
+/*!
+ * Copyright 2018 by Contributors
+ * \author Rory Mitchell
+ */
+
+#include <xgboost/linear_updater.h>
+#include "coordinate_common.h"
+#include "../common/timer.h"
+
+namespace xgboost {
+namespace linear {
+
+DMLC_REGISTRY_FILE_TAG(updater_coordinate);
+
+// training parameter
+struct CoordinateTrainParam : public dmlc::Parameter<CoordinateTrainParam> {
+  /*! \brief learning_rate */
+  float learning_rate;
+  /*! \brief regularization weight for L2 norm */
+  float reg_lambda;
+  /*! \brief regularization weight for L1 norm */
+  float reg_alpha;
+  std::string coordinate_selection;
+  float maximum_weight;
+  int debug_verbose;
+  // declare parameters
+  DMLC_DECLARE_PARAMETER(CoordinateTrainParam) {
+    DMLC_DECLARE_FIELD(learning_rate)
+        .set_lower_bound(0.0f)
+        .set_default(1.0f)
+        .describe("Learning rate of each update.");
+    DMLC_DECLARE_FIELD(reg_lambda)
+        .set_lower_bound(0.0f)
+        .set_default(0.0f)
+        .describe("L2 regularization on weights.");
+    DMLC_DECLARE_FIELD(reg_alpha)
+        .set_lower_bound(0.0f)
+        .set_default(0.0f)
+        .describe("L1 regularization on weights.");
+    DMLC_DECLARE_FIELD(coordinate_selection)
+        .set_default("cyclic")
+        .describe("Coordinate selection algorithm");
+    DMLC_DECLARE_FIELD(maximum_weight)
+        .set_lower_bound(0.0f)
+        .set_default(1e15f)
+        .describe(
+            "Maximum absolute value of linear model weights. Prevents "
+            "coefficients increasing to infinity in separable logistic "
+            "regression problems.");
```

Comment:
I'd rather have an error thrown when complete separation happens and things go out of wack. I can't think of any practical context when such clipping could be useful, as the regularization is a more proper tool to deal with this.